src/targets/crates.ts

```diff
@@ -215,18 +222,15 @@ export class CratesTarget extends BaseTarget {
     };
 
     logger.info(`Publishing ${crate.name}`);
-    for (let i = 0; i <= 1; i++) {
-      try {
-        return await spawnProcess(CARGO_BIN, args, { env });
-      } catch (e) {
-        if (i === 0 && e.message.includes(VERSION_ERROR)) {
-          logger.warn(
-            `Potential stale cache detected while publishing ${crate.name}, trying again...`
-          );
-          continue;
-        } else {
-          throw e;
-        }
+    try {
+      return await spawnProcess(CARGO_BIN, args, { env });
+    } catch (e) {
+      if (e.message.includes(VERSION_ERROR)) {
+        logger.warn(`Potential stale cache detected, trying again...`);
+        await sleepAsync(RETRY_DELAY_MS);
+        return spawnProcess(CARGO_BIN, args, { env });
```

Comment:
hah, skipping `await`, very clever optimization :)

Reply:
It's closer to the original. However, I think I'll have to revert this and really spin for multiple times + increase the timeout. No idea what the root cause of this is.

Reply:
Maybe one of these help:
 - https://github.com/rust-lang/cargo/issues/8273
 - https://github.com/rust-lang/cargo/issues/3377#issuecomment-361819305

Reply:
Thanks, I tried that as well. Unfortunately, you can see in the above output that the crates index was in fact refreshed, but still didn’t contain the new version.

This leads me to believe that it’s server side caches that interfer with this, and the only way around would be larger delays and more retries.

Reply:
What if we kept polling the API directly until we get a response (ie `https://crates.io/api/v1/crates/symbolic/7.3.5`)